File: src/config/coreConfig.ts
import {BaseConfig} from './baseConfig.js';
import IManagerBot from '../interfaces/iManagerBot.js';
import {SupportedModels, SupportedProviders} from '../interfaces/literalTypes.js';
import {Snowflake} from 'discord.js';

interface CoreConfigSchema {
    managerBot: IManagerBot;
    model: SupportedModels;
    provider: SupportedProviders;
    aiToken: string;
    idleIntervalSec: number;
    roleplayChannels: Snowflake[];
}

class CoreConfig extends BaseConfig<CoreConfigSchema> {
    constructor() {
        const defaultConfig: CoreConfigSchema = {
            managerBot: {
                botToken: ''
            },
            model: 'gpt-4o',
            provider: 'openai',
            aiToken: '',
            idleIntervalSec: 180,
            roleplayChannels: []
        };

        super('core-config.json', defaultConfig);
    }
}

export default CoreConfig;

File: dotnet/ChatRoleplay/Config/CoreConfig.cs
using System.Text.Json.Serialization;

namespace ChatRoleplay.Config;

public class ManagerBotConfig
{
    [JsonPropertyName("botToken")]
    public string BotToken { get; set; } = string.Empty;
}

public class CoreConfig
{
    [JsonPropertyName("managerBot")]
    public ManagerBotConfig ManagerBot { get; set; } = new();

    [JsonPropertyName("model")]
    public string Model { get; set; } = "gpt-4o";

    [JsonPropertyName("provider")]
    public string Provider { get; set; } = "openai"; // "openai" or "ollama"

    [JsonPropertyName("customEndpoint")]
    public string? CustomEndpoint { get; set; }

    [JsonPropertyName("aiToken")]
    public string AiToken { get; set; } = string.Empty;

    [JsonPropertyName("idleIntervalSec")]
    public int IdleIntervalSec { get; set; } = 180;

    [JsonPropertyName("roleplayChannels")]
    public List<ulong> RoleplayChannels { get; set; } = [];

    [JsonPropertyName("useToolCalling")]
    public bool UseToolCalling { get; set; } = true;
}
